diff --git a/config.sub b/config.sub
index 38f3d037a78..d285431845d 100755
--- a/config.sub
+++ b/config.sub
@@ -1749,7 +1749,7 @@ case $os in
 	     | onefs* | tirtos* | phoenix* | fuchsia* | redox* | bme* \
 	     | midnightbsd* | amdhsa* | unleashed* | emscripten* | wasi* \
 	     | nsk* | powerunix* | genode* | zvmoe* | qnx* | emx* | zephyr* \
-	     | fiwix* )
+	     | fiwix* | quark* )
 		;;
 	# This one is extra strict with allowed versions
 	sco3.2v2 | sco3.2v[4-9]* | sco5v6*)
diff --git a/gcc/config.gcc b/gcc/config.gcc
index 087aaa7be37..f82dfd4449b
--- a/gcc/config.gcc
+++ b/gcc/config.gcc
@@ -1962,6 +1962,9 @@ i[34567]86-*-elf*)
 x86_64-*-elf*)
 	tm_file="${tm_file} i386/unix.h i386/att.h elfos.h newlib-stdint.h i386/i386elf.h i386/x86-64.h"
 	;;
+x86_64-*-quark*)
+	tm_file="${tm_file} i386/unix.h i386/att.h elfos.h newlib-stdint.h i386/i386elf.h i386/x86-64.h quark.h"
+	;;
 x86_64-*-rtems*)
 	tm_file="${tm_file} i386/unix.h i386/att.h elfos.h newlib-stdint.h i386/i386elf.h i386/x86-64.h i386/rtemself.h rtems.h"
 	;;
diff --git a/libgcc/config.host b/libgcc/config.host
index 6a88ee5a2dd..1d7c0fc79e7
--- a/libgcc/config.host
+++ b/libgcc/config.host
@@ -756,7 +756,7 @@ i[34567]86-*-elfiamcu)
 i[34567]86-*-elf*)
 	tmake_file="$tmake_file i386/t-crtstuff t-crtstuff-pic t-libgcc-pic"
 	;;
-x86_64-*-elf* | x86_64-*-rtems*)
+x86_64-*-elf* | x86_64-*-quark* | x86_64-*-rtems*)
 	tmake_file="$tmake_file i386/t-crtstuff t-crtstuff-pic t-libgcc-pic"
 	case ${host} in
 	  x86_64-*-rtems*)
diff --git a/libgcc/configure b/libgcc/configure
index 18418336c9b..422298a285b 100755
--- a/libgcc/configure
+++ b/libgcc/configure
@@ -2712,7 +2712,7 @@ if test "${enable_gcov+set}" = set; then :
   enableval=$enable_gcov;
 else
   case $target in
-       bpf-*-*)
+       bpf-*-* | *-quark*)
          enable_gcov=no
        ;;
        *)
diff --git a/libgcc/configure.ac b/libgcc/configure.ac
index 85e4f1bc48b..0eb58a8500b
--- a/libgcc/configure.ac
+++ b/libgcc/configure.ac
@@ -161,7 +161,7 @@ GCC_TOPLEV_SUBDIRS
 AC_ARG_ENABLE(gcov,
 [  --disable-gcov          don't provide libgcov and related host tools],
 [], [case $target in
-       bpf-*-*)
+       bpf-*-* | *-quark*)
          enable_gcov=no
        ;;
        *)
